kafka-client/src/main/java/dev/responsive/kafka/internal/stores/ResponsiveKeyValueStore.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright 2023 Responsive Computing, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.responsive.kafka.internal.stores;
+
+import static org.apache.kafka.streams.processor.internals.ProcessorContextUtils.asInternalProcessorContext;
+
+import dev.responsive.kafka.api.stores.ResponsiveKeyValueParams;
+import dev.responsive.kafka.internal.config.InternalConfigs;
+import dev.responsive.kafka.internal.db.CassandraTableSpecFactory;
+import dev.responsive.kafka.internal.db.spec.CassandraTableSpec;
+import dev.responsive.kafka.internal.utils.TableName;
+import java.util.List;
+import java.util.concurrent.TimeoutException;
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.common.utils.LogContext;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.errors.ProcessorStateException;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.processor.StateStore;
+import org.apache.kafka.streams.processor.StateStoreContext;
+import org.apache.kafka.streams.processor.internals.Task.TaskType;
+import org.apache.kafka.streams.query.Position;
+import org.apache.kafka.streams.state.KeyValueIterator;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.state.internals.StoreQueryUtils;
+import org.slf4j.Logger;
+
+public class ResponsiveKeyValueStore implements KeyValueStore<Bytes, byte[]> {
+
+  private final Logger log;
+
+  private final ResponsiveKeyValueParams params;
+  private final TableName name;
+  private final Position position; // TODO(IQ): update the position during restoration
+
+  private boolean open;
+  private KeyValueOperations operations;
+  private ResponsiveStoreRegistry storeRegistry;
+  private CassandraTableSpec spec;
+  private StateStoreContext context;
+
+  public ResponsiveKeyValueStore(final ResponsiveKeyValueParams params) {
+    this.params = params;
+    this.name = params.name();
+    this.position = Position.emptyPosition();
+
+    log = new LogContext(
+        String.format("store [%s] ", name.kafkaName())
```

Comment:
nit: probably worth including the spec and/or operations type/name in the log prefix

Reply:
since we don't know this in the constructor I made the logger non-final, and the re-initialized it when `init` is called. I believe there's no logs yet pre-`init`, but this could lead to weird behavior down the road